lib/rules/no-curly-component-invocation.js

```diff
@@ -47,14 +49,29 @@ const BUILT_INS = new Set([
 
 const ALWAYS_CURLY = new Set(['yield']);
 
+function toAngleAttrValue(attr) {
+  const node = attr.value;
+  if (node.type === 'SubExpression') {
+    return b.mustache(node.path, node.params, node.hash);
+  }
```

Comment:
Can you check for `concat` here, and handle it specially?

Specifically adding a test for

```hbs
{{#some-thing foo=(concat 'lol-' something ' ' 'wtf')}}
{{/some-thing}}
```

emitting:

```hbs
<SomeThing @foo="lol-{{something}} wtf">
</SomeThing>
```

Reply:
@rwjblue looks like we have printer issue
![image](https://user-images.githubusercontent.com/1360552/105413150-c0f9e200-5c46-11eb-981c-4aee87ebaed7.png)

```hbs
<SomeThing @foo={{concat 'lol-' something ' ' 'wtf'}}>
</SomeThing>
```

should be fine for now

Reply:
my bad, 
![image](https://user-images.githubusercontent.com/1360552/105413999-f5ba6900-5c47-11eb-8167-9e83ce61fd7c.png) works fine

Reply:
* fixed with `isValueLess` toggling